### ISSUE TITLE
No Replace when using Regular expression with lookahead tokens at the begining of Find what requests

### ISSUE THREAD

**Issue by Navi1982** [notepad-plus-plus/notepad-plus-plus#4278]
### Description of the Issue

Replace button don't replace found matches in case when using regular expression with tokens that look ahead, i.e. like \b  or (?=...) or similar tokens that dont capture matches. It happens when mentioned tokens inserted at the beginning of "Find what" requests.

A requested matches are found, but are not replacing.

I've used regular expression as below:
\b\r\n(?!(\r\n)+)
to replace with " " (a space symbol) to concatenate lines with single CrLf combination right after word boundary. I know about existence of much perfect regular expressions for that. BUT, there is an issue.
### Steps to Reproduce the Issue

1.  Create any text file with text separated by CrLf, after each aprox 80th symbols and CrLfCrLf sequences between paragraphs.
2. Make a request in "Find what:" edit box as \b\r\n and put a space in "Replace with:" edit box.
3. Play with "Find Next" and "Replace" buttons. You'll find that matches are found, but are not replacing.

### Expected Behavior

I expecting that empty matches at the begining of request is ignored.

### Actual Behavior

Similar things I've read in other issues, but can't remember their numbers.
### Debug Information

Used last version v7.5.5 of Notepad++

**Comment by sasumner** [comment]
@Navi1982 Can you provide specific examples for "(?=...) or similar tokens that dont capture matches"?

**Comment by Navi1982** [comment]
@sasumner I think a good example with same meaning will be:
(?=[a-zA-Z0-9_]+)\r\n
is equivalent with:
\b\r\n

**Comment by sasumner** [comment]
@Navi1982 I don't think that example works.  It is asserting that the next thing to be found will be a "word character", but then the next thing is a `\r` which isn't a word char.  So, never matches.

I do see that replacing `\b\r\n` with any literal text doesn't happen even though matches are found (interesting, indeed!), but since you hinted that there were other scenarios that worked like this I was interested in what they were.

**Comment by sasumner** [comment]
More experimenting:  
Doing a replace-all on `\b\r\n` with some literal text is successful.
Doing an interactive replace on `\>\r\n` with some literal text is not successful (same result as `\b\r\n').
Doing a replace-all on `\>\r\n` with some literal text is successful.
What does it all mean? :)

**Comment by Navi1982** [comment]
Yes. I confirm that using replace all work fine. But, the use of "replace all" is good when you are sure of the accuracy of a regular expression. My last query was modified like below:
\b(\r\n)(?!(\r\n)+)
...and work as before: only on "replace all". But need to work with "replace"

**Comment by MAPJe71** [comment]
Yes there are issues, see [this Community post](https://notepad-plus-plus.org/community/topic/13513/proximity-search/12) by `guy038`.
